### PR TITLE
docs: extract how-to guides from "Server" page

### PR DESCRIPTION
**Pull request by bajtos**
As part of our move towards four-quadrant documentation system (see #5549 and #5718), we (@hacksparrow and me) took a look at restructuring [Server](https://github.com/strongloop/loopback-next/blob/d653cfc4c96c78282e408b5f7054eee04a67841c/docs/site/Server.md) page and extracted two new how-to guides:

- Enabling HTTPS (under Deployment)
- Customizing how OpenAPI spec is served (under a new section REST APIs)

The goal of this work is to explore how to best (re)structure the existing content, so that we can create better user-stories. 

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- Affected artifact templates in `packages/cli` were updated
- Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/guides/rest/rest-api-guides.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+lang: en
+title: 'REST API Guides'
+keywords: LoopBack 4.0, LoopBack 4, Node.js, REST
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/rest-api-guides.html
+---
```
Comment (bajtos) [inline docs/site/guides/rest/rest-api-guides.md:7]: Let's remove this empty page after we land https://github.com/strongloop/loopback.io/pull/996

File: docs/site/sidebars/lb4_sidebar.yml [hunk 2]
```diff
@@ -177,11 +177,19 @@ children:
         url: core-tutorial-part11.html
         output: 'web, pdf'
 
-- title: 'Usage Scenarios'
+- title: 'How-to guides'
```
Comment (bajtos) [inline docs/site/sidebars/lb4_sidebar.yml:180]: I think this change should be extracted into a standalone PR, where we should also update `Usage-senarios.html` to something like `how-to-guides.html` while keeping the old URL as an alias to avoid broken (404) links.

File: docs/site/sidebars/lb4_sidebar.yml [hunk 3]
```diff
@@ -177,11 +177,19 @@ children:
...
   url: Usage-scenarios.html
   output: 'web, pdf'
   children:
 
+  - title: 'REST APIs'
```
Comment (bajtos) [inline docs/site/sidebars/lb4_sidebar.yml:185]: Discussion point: should we call this section "REST APIs" or something like "Building REST APIs"?

IMO, "REST APIs" could be interpreted as covering both building and consuming APIs, while this section of our docs is covering the building part only. For consumption, we already have the page "Calling other APIs and web services" that's outside of "REST APIs".

Comment (hacksparrow) [reply docs/site/sidebars/lb4_sidebar.yml:185]: Since it's from a LoopBack context, "REST APIs" in that particular location should be understood as configuring/building REST APIs.

File: docs/site/Server.md [hunk 4]
```diff
@@ -69,40 +69,6 @@ const app = new RestApplication({
 });
 ```
 
-### Customize How OpenAPI Spec is Served
```
Comment (bajtos) [inline docs/site/Server.md:72]: In order to make it easier to find related guides when reading "Server" explanation page, perhaps we should add a new section here with links to those guides. I am thinking about something along the following lines:

```markdown
## Common tasks

- [Customize how OpenAPI spec is served](rest-customize-openapi-serving.html)
- [Enable HTTPS](deployment-guides-enabling-https.html)
```

File: docs/site/guides/rest/customize-openapi.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+---
+lang: en
+title: 'Customizing how OpenAPI spec is served'
+keywords: LoopBack 4.0, LoopBack 4, Node.js, OpenAPI, Customize
```
Comment (dhmlau) [inline docs/site/guides/rest/customize-openapi.md:4]: Maybe `Customize` -> `Customization`? (it seems like our keywords are nouns, so might want to keep it consistent)

File: docs/site/guides/rest/rest-api-guides.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+lang: en
+title: 'REST API Guides'
+keywords: LoopBack 4.0, LoopBack 4, Node.js, REST
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/rest-api-guides.html
```
Comment (dhmlau) [inline docs/site/guides/rest/rest-api-guides.md:6]: Similar comment: 

```suggestion
permalink: /doc/en/lb4/Rest-api-guides.html
```

File: docs/site/guides/rest/customize-openapi.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/rest-customize-openapi-serving.html
```
Comment (dhmlau) [inline docs/site/guides/rest/customize-openapi.md:6]: As @raymondfeng stated in the other PRs, we might want to have the file name to begin with capital letter:
```suggestion
permalink: /doc/en/lb4/Rest-customize-openapi-serving.html
```

File: docs/site/guides/rest/rest-api-guides.md (hunk 6 above)
Comment (bajtos) [reply docs/site/guides/rest/rest-api-guides.md:6]: I would like to transition our URL scheme to all-lowercase names and eventually to nested paths (`/reference/reserved-binding-keys.html`) to match the directory layout in `docs/site`. 

I encountered some issues when using nested paths, so I am using `-` instead of `/` for now, and also slightly reordering the order of components for better readability:

```
guides/rest-api.md ->rest-api-guides.html
```

We are already following the all-lowercase slug style in migration guide, see e.g. here:

https://github.com/strongloop/loopback-next/blob/ae6427322451c914ae54f44dbb656981e7fbbb81/docs/site/migration/datasources.md#L7

Let's discuss in #5798

File: docs/site/sidebars/lb4_sidebar.yml (hunk 2 above)
Comment (bajtos) [reply docs/site/sidebars/lb4_sidebar.yml:180]: The page "Usage-scenarios.html" was removed by https://github.com/strongloop/loopback-next/pull/5773.

I opened  #5799 to rename the sidebar section.

File: docs/site/sidebars/lb4_sidebar.yml (hunk 3 above)
Comment (bajtos) [reply docs/site/sidebars/lb4_sidebar.yml:185]: I'll go with "Building REST APIs" then.

File: docs/site/guides/rest/rest-api-guides.md (hunk 6 above)
Comment (bajtos) [reply docs/site/guides/rest/rest-api-guides.md:6]: Based on discussion with @raymondfeng, we will stick with Sentence-style.html to land my PRs and then normalize URL slugs afterwards, as this part may take a bit longer to reach consensus.

**Comment by bajtos** [comment]
Rebased on top of the latest master and resolved review comments & todo items.

The PR is ready for final review and landing.

@raymondfeng @dhmlau @hacksparrow PTAL

File: docs/site/guides/deployment/enabling-https.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
+---
+lang: en
+title: 'Enabling HTTPS'
+keywords: LoopBack 4.0, LoopBack 4, Node.js, HTTPS, Deployment
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Enabling-https.html
+---
+
+Enabling HTTPS for the LoopBack REST server is just a matter of specifying the
+protocol as `https` and specifying the credentials.
```
Comment (hacksparrow) [inline docs/site/guides/deployment/enabling-https.md:10]: "...specifying the credentials." -> "...specifying the credentials in the rest options of the application configuration."?

**Comment by hacksparrow** [comment]
Not a biggie, left a suggestion. LGTM.

File: docs/site/guides/deployment/enabling-https.md (hunk 8 above)
Comment (bajtos) [reply docs/site/guides/deployment/enabling-https.md]: Sounds good, addressed in 9871197.
